src/logic/calculate.js

```diff
@@ -1,7 +1,59 @@
-import Big from 'big.js';
+import operate from './operate';
 
-const calculate = () => {
+const calculate = (calculatorData, buttonName) => {
+  const { total, next, operation } = calculatorData;
+  const result = calculatorData;
+  const numbers = /[0-9]/;
+  const operators = ['+', '-', 'x', '÷', '%'];
 
-};
+  if (buttonName === 'AC') {
+    result.total = undefined;
+    result.next = undefined;
+    result.operation = undefined;
+  }
+
+  if (buttonName === '.') {
+    if (!total && !next) {
+      result.total = '0.';
+    }
+    if (total && next && total.indexOf('.') === -1) {
+      result.total = '{total}.';
+    }
+    if (next && next.indexOf('.') === -1) {
+      result.next = '{next}.';
+    }
+  }
+
+  if (buttonName === '+/-') {
+    if (total && !next) {
+      result.total = operate(total, -1, 'x');
+    }
+    if (next) {
+      result.total = operate(next, -1, 'x');
+    }
+  }
 
+  if (operators.includes(buttonName) /* && buttonName === operation */) {
```

Comment:
Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.